cmd/upgrade/cluster/cmd.go

```diff
@@ -428,6 +430,31 @@ func run(cmd *cobra.Command, _ []string) {
 		os.Exit(1)
 	}
 
+	// check if the cluster upgrade requires gate agreements
+	gates, err := ocmClient.GetMissingGateAgreements(cluster.ID(), upgradePolicy)
+	if err != nil {
+		reporter.Errorf("Failed to check for missing gate agreements upgrade for cluster '%s': %v", clusterKey, err)
+		os.Exit(1)
+	}
+
+	for _, gate := range gates {
+		reporter.Infof("Gate: %v", gate)
+		if !gate.STSOnly() {
+			// for non sts gates we require user agreement
+			if !confirm.Prompt(true,
+				"I acknowledge that my workloads are no longer using removed APIs. (Background:'%s)",
```

Comment:
this message is specific to OCP 4.9. When 4.10 comes around and we add (via configuration) a gate for something completely different, this message makes no sense.

Why not just hardcode the entire description if we're hardcoding half of it here?

Better to get this string from the `gates[]` imo

Reply:
I agree with @gregsheremeta - the API already returns the correct messaging on purpose so that the CLI does not need to know the context for the ACK (apart from the STS one)